### DISCUSSION TITLE
OpenDDSharp Standard is coming

### DISCUSSION THREAD

**Discussion by jmmorato** [jmmorato/openddsharp#77] [General]
Hi there,

I recently published new NuGet packages that contain a new wrapper for .netstandard:

| Package | NuGet |
|---------|-------|
| [OpenDDSharp.Standard][OpenDDSharpNuget] | [![OpenDDSharpShield]][OpenDDSharpNuget] |
| [OpenDDSharp.Standard.IdlGenerator][OpenDDSharpIdlGeneratorNuget] | [![OpenDDSharpIdlGeneratorShield]][OpenDDSharpIdlGeneratorNuget] |
| [OpenDDSharp.Native][OpenDDSharpNativeNuget] | [![OpenDDSharpNativeShield]][OpenDDSharpNativeNuget] |
| [OpenDDSharp.Templates][OpenDDSharpTemplatesNuget] | [![OpenDDSharpTemplatesShield]][OpenDDSharpTemplatesNuget] |

[OpenDDSharpNuget]: https://www.nuget.org/packages/OpenDDSharp.Standard/
[OpenDDSharpShield]: https://img.shields.io/nuget/v/OpenDDSharp.Standard.svg
[OpenDDSharpIdlGeneratorNuget]: https://www.nuget.org/packages/OpenDDSharp.Standard.IdlGenerator/
[OpenDDSharpIdlGeneratorShield]: https://img.shields.io/nuget/v/OpenDDSharp.Standard.IdlGenerator.svg
[OpenDDSharpNativeNuget]: https://www.nuget.org/packages/OpenDDSharp.Native/
[OpenDDSharpNativeShield]: https://img.shields.io/nuget/v/OpenDDSharp.Native.svg
[OpenDDSharpTemplatesNuget]: https://www.nuget.org/packages/OpenDDSharp.Templates/
[OpenDDSharpTemplatesShield]: https://img.shields.io/nuget/v/OpenDDSharp.Templates.svg

This wrapper is meant to replace the current CPP/CLI wrapper but still a long way to go. The API is not fully completed yet (but contains all the entities and the most important classes to use in OpenDDSharp) the code generation for linux need to be improved, performance test and memory test implemented, I plan to migrate to new platforms (currently win-x86, win-64 and linux-x64 are supported)...

Even if you can create applications for linux, you still need to work with Visual Studio. I'll create some documentation for the new wrapper soon, for the moment you can check the hello_world project I published here: https://github.com/jmmorato/openddsharp_standard_helloworld

**Comment by rclendan** [comment]
Just wondering what the timeline is for VS 2022 support.

**Reply by jmmorato** [reply]
To compile OpenDDSharp, OpenDDS should be compatible. Latest version of OpenDDS have support for VS2022, so next version of OpenDDSharp will have it too.

To develop with OpenDDSharp and the standard wrapper, the support is already included. Have you tried? if so, what kind of problem did you find?

In any case, the new version of OpenDDSharp will arrive around on Q2 2022. Unfortunately, I have not much time for the project during Q1.
